### PR TITLE
Don't allow picking on the sunPostProcess framebuffer.

### LINKED ISSUE #5402: Entity Picking Bug
A coworker was working on something and came across a weird issue when attempting to pick entities in Cesium. I've put together a little example [here](http://cesiumjs.org/Cesium/Apps/Sandcastle/?src=Hello%20World.html&label=Showcases&gist=714cdba64c65b7cf0779cdc88f957d77)

I attempted to post this in the forums but they weren't showing up.

The basic idea is, we are using the mouse_move event to pick and see what entities are under the cursor and display a tooltip for that entity. The sandcastle example above does just that.  We've come across a repeatable issue where Cesium doesn't correctly report the picked objects.

We found this in Cesium 1.32 I believe but have been able to reproduce it in the Sandcastle as well which I believe is currently at 1.33.

To reproduce the issue:
1. Zoom in to the right of the ISS path polyline (the purple one)
2. Adjust the camera angle downward (as low as possible) so that you're looking at the horizon. You should be below the polyline now
3. Click, drag and rotate the globe to a point beyond the ISS polyline (now on the left side of it)
4. Rotate the camera back around and look up at the ISS polyline
5. Mouseover the polyline for the ISS

If the steps above are executed correctly, the tooltip won't be displayed. In testing, it appears that the entity isn't correctly picked. Our code we were testing with would console log the value of the variable that we stored the picked entity into and in this instance, it would return undefined. 

Getting it to fail my take a few tries but it's something that we've definitely been able to replicate numerous times. Is this a known issue?

### LINKED ISSUE #5478: Picking and drill-picking don't work after rotating the globe
If you spin the globe to the opposite side (Australia, for example) and then come back to the US, picking and drill-picking stop working correctly.

Picking:

pickingproblem

And also, drill-picking:

drillpickproblem

/cc @AnneWoepse

### PR DESCRIPTION
**Pull request by emackey**
Fixes #5478.

Likely fixes #5402.

/cc @bagnell @AnneWoepse

@bagnell I'm curious about the `if` block I'm modifying here.  It has two `else if` clauses which could possibly both be false.  What happens if all three paths go false at the same time?  Seems like the framebuffer would not be assigned, would it have a reasonable default?

### CODE REVIEW

**Comment by pjcozzi** [comment]
Please update CHANGES.md

**Comment by emackey** [comment]
@bagnell Also, it's concerning that picking would "stay broken" even after this code switched back to using the globeDepth framebuffer.  Is switching framebuffers here somehow unsafe for picking?

**Comment by bagnell** [comment]
>What happens if all three paths go false at the same time? Seems like the framebuffer would not be assigned, would it have a reasonable default?

It will default to whatever the pass state framebuffer is. When picking, it will be the picking framebuffer. If the pass state framebuffer is undefined, it'll render to the default framebuffer (directly to canvas).

>Also, it's concerning that picking would "stay broken" even after this code switched back to using the globeDepth framebuffer. Is switching framebuffers here somehow unsafe for picking?

Yes, during the pick pass it should never switch framebuffers. All of the post processing and globe depth copying should be disabled when picking.

File: Source/Scene/Scene.js [hunk 1]
```diff
@@ -2468,7 +2468,7 @@ define([
             scene._fxaa.clear(context, passState, clearColor);
         }
 
-        if (environmentState.isSunVisible && scene.sunBloom && !useWebVR) {
+        if (environmentState.isSunVisible && scene.sunBloom && !useWebVR && !picking) {
```
Comment (bagnell) [inline Source/Scene/Scene.js]: I'm not sure this is the right fix. The condition should always evaluate to false when picking. All of the environment should not render and all post processing should be disabled when `frameState.pick` is `true`.

Comment (emackey) [inline Source/Scene/Scene.js]: Maybe the environment isn't being updated correctly?  The commit that caused this (a9a275f) changes the location of the environment update.

Comment (emackey) [inline Source/Scene/Scene.js]: And, notably, the offending commit didn't touch the line I've changed here.  It remotely changed the results of this `if` check.

**Comment by emackey** [comment]
> All of the post processing and globe depth copying should be disabled when picking.

During my testing, I added `console.log` statements to the various `if` clauses here, and the globe depth one (the middle one) does get executed when the Sun post process filter isn't in effect.  Maybe this whole section of code needs stronger protection from picking.

**Comment by bagnell** [comment]
>Maybe the environment isn't being updated correctly?

I think it might be updating correctly but out of order? If you look at `Sun.update` you'll see this block of code:
```javascript
if (!frameState.passes.render) {
    return undefined;
}
```
That would cause `environmentState.isSunVisible` to be false in that condition.

>During my testing, I added console.log statements to the various if clauses here, and the globe depth one (the middle one) does get executed when the Sun post process filter isn't in effect. Maybe this whole section of code needs stronger protection from picking.

That's strange, if you look several lines up you'll see this:
`var useGlobeDepthFramebuffer = environmentState.useGlobeDepthFramebuffer = !picking && defined(scene._globeDepth);`

**Comment by emackey** [comment]
> That's strange

Oh, I didn't limit my console logging to just the pick pass, fixed now.  So yes, in master, the pick pass chooses either Sun bloom (the first clause, when the Sun is visible in the frame), or it falls past all three conditionals to leave the framebuffer alone (skipping over globe depth and fxaa).

**Comment by bagnell** [comment]
Ah, OK, I found the real issue. The environment state is never updated on a pick. We simply need to call `updateEnvironment` in `Scene.pick` right before `updateAndExecuteCommands`. Do you you want to submit the fix?

**Comment by emackey** [comment]
Actually I added the call in two places, is this OK?

**Comment by bagnell** [comment]
Yup, thanks @emackey!

**Pull request merged**